Library/Homebrew/formula_installer.rb

```diff
@@ -601,6 +601,10 @@ def install_dependency(dep, inherited_options)
   rescue FormulaInstallationAlreadyAttemptedError
     # We already attempted to install f as part of the dependency tree of
     # another formula. In that case, don't generate an error, just move on.
+    ignore_interrupts do
+      tmp_keg.rename(installed_keg) if tmp_keg && !installed_keg.directory?
+      linked_keg.link if keg_was_linked
+    end
     nil
   rescue Exception # rubocop:disable Lint/RescueException
```

Comment:
```suggestion
  rescue Exception => e # rubocop:disable Lint/RescueException
```

and then below `raise unless e.is_a?(FormulaInstallationAlreadyAttemptedError)`

Reply:
good call. I made these changes and squashed into 096a5bd